pkg/util/network/network.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1"
+
+	"k8s.io/utils/net"
+)
+
+// IsIPv4OnlyCluster returns true if the cluster networking is IPv4-only.
+func IsIPv4OnlyCluster(cluster *kubermaticv1.Cluster) bool {
+	return len(cluster.Spec.ClusterNetwork.Pods.CIDRBlocks) == 1 && net.IsIPv4CIDRString(cluster.Spec.ClusterNetwork.Pods.CIDRBlocks[0])
+}
+
+// IsIPv6OnlyCluster returns true if the cluster networking is IPv6-only.
+func IsIPv6OnlyCluster(cluster *kubermaticv1.Cluster) bool {
+	return len(cluster.Spec.ClusterNetwork.Pods.CIDRBlocks) == 1 && net.IsIPv6CIDRString(cluster.Spec.ClusterNetwork.Pods.CIDRBlocks[0])
+}
+
+// IsDualStackCluster returns true if the cluster networking is dual-stack (IPv4 + IPv6).
```

Comment:
nit: There should be away to get this error. If there are typos in the cider spec it will be easier to debug with the error.

Here is example with typo hard to spot with eyes.
 
`CIDRBlocks: []string{"172.25.0.0/16", "2020:9108:d40:e04:f5dd:7887:01h4:3070/104"},`
`


`failed to parse cidr value:"2020:9108:d40:e04:f5dd:7887:01h4:3070/104" with error:invalid CIDR address: 2020:9108:d40:e04:f5dd:7887:01h4:3070/104`

Reply:
I intentionally decided to not return error from these function because:
 - CIDRs are validated in the validation webhook. That prevents from using invalid CIDRs in the cluster spec. If it was invalid at the place where these functions are used, we would have a big big problem anyway.
 - I expect these funcs to spread across our code base quite rapidly :) Checking for error every time they are used would result in code bloat and lower readability.